SecurityPkg/Tcg/Opal/OpalPassword/OpalDriver.c

```diff
@@ -2793,43 +2802,70 @@ OpalEfiDriverBindingStart(
     Controller,
     EFI_OPEN_PROTOCOL_BY_DRIVER
     );
+
+  if (Status == EFI_UNSUPPORTED)
+    BlkIo = NULL;
+
   if (EFI_ERROR(Status)) {
-    //
-    // Block_IO not supported on handle
-    //
-    if(Status == EFI_UNSUPPORTED) {
-      BlkIo = NULL;
-    } else {
+    Status2 = gBS->OpenProtocol(

```

Comment:
Should EFI_BLOCK_IO2_PROTOCOL not take priority over EFI_BLOCK_IO_PROTOCOL?

Reply:
UEFI spec does not specify which protocol has priority. I have checked all drivers that produce BLOCK_IO or BLOCK_IO and it looks like all driver produce both, so it shouldn't matter. All we need the BLOCK_IO for is the MediaId, which shouldn't be different depending on BLOCK_IO or BLOCK_IO2 @mkopec

Reply:
Ok, thanks.